tests/e2e/tracehandler_testdata/democracy.json

```diff
@@ -0,0 +1 @@
ob"},"State":{"democ":{"ValBalances":null,"ValPowers":{"alice":511,"bob":500,"carol":500},"RepresentativePowers":null,"Params":null,"Rewards":null,"ConsumerChains":null,"AssignedKeys":null,"ProviderKeys":null,"ConsumerChainQueueSizes":null,"GlobalSlashQueueSize":null,"RegisteredConsumerRewardDenoms":null,"Proposals":null},"provi":{"ValBalances":null,"ValPowers":{"alice":511,"bob":500,"carol":500},"RepresentativePowers":null,"Params":null,"Rewards":null,"ConsumerChains":null,"AssignedKeys":null,"ProviderKeys":null,"ConsumerChainQueueSizes":null,"GlobalSlashQueueSize":null,"RegisteredConsumerRewardDenoms":null,"Proposals":null}}},{"ActionType":"main.relayPacketsAction","Action":{"ChainA":"provi","ChainB":"democ","Port":"provider","Channel":0},"State":{"democ":{"ValBalances":null,"ValPowers":{"alice":511,"bob":500,"carol":500},"RepresentativePowers":null,"Params":null,"Rewards":null,"ConsumerChains":null,"AssignedKeys":null,"ProviderKeys":null,"ConsumerChainQueueSizes":null,"GlobalSlashQueueSize":null,"RegisteredConsumerRewardDenoms":null,"Proposals":null},"provi":{"ValBalances":null,"ValPowers":{"alice":511,"bob":0,"carol":500},"RepresentativePowers":null,"Params":null,"Rewards":null,"ConsumerChains":null,"AssignedKeys":null,"ProviderKeys":null,"ConsumerChainQueueSizes":null,"GlobalSlashQueueSize":null,"RegisteredConsumerRewardDenoms":null,"Proposals":null}}},{"ActionType":"main.relayPacketsAction","Action":{"ChainA":"provi","ChainB":"democ","Port":"provider","Channel":0},"State":{"democ":{"ValBalances":null,"ValPowers":{"alice":511,"bob":0,"carol":500},"RepresentativePowers":null,"Params":null,"Rewards":null,"ConsumerChains":null,"AssignedKeys":null,"ProviderKeys":null,"ConsumerChainQueueSizes":null,"GlobalSlashQueueSize":null,"RegisteredConsumerRewardDenoms":null,"Proposals":null}}},{"ActionType":"main.unjailValidatorAction","Action":{"Provider":"provi","Validator":"bob"},"State":{"democ":{"ValBalances":null,"ValPowers":{"alice":511,"bob":0,"carol":500},"RepresentativePowers":null,"Params":null,"Rewards":null,"ConsumerChains":null,"AssignedKeys":null,"ProviderKeys":null,"ConsumerChainQueueSizes":null,"GlobalSlashQueueSize":null,"RegisteredConsumerRewardDenoms":null,"Proposals":null},"provi":{"ValBalances":null,"ValPowers":{"alice":511,"bob":500,"carol":500},"RepresentativePowers":null,"Params":null,"Rewards":null,"ConsumerChains":null,"AssignedKeys":null,"ProviderKeys":null,"ConsumerChainQueueSizes":null,"GlobalSlashQueueSize":null,"RegisteredConsumerRewardDenoms":null,"Proposals":null}}},{"ActionType":"main.relayPacketsAction","Action":{"ChainA":"provi","ChainB":"democ","Port":"provider","Channel":0},"State":{"democ":{"ValBalances":null,"ValPowers":{"alice":511,"bob":500,"carol":500},"RepresentativePowers":{"alice":100500000,"bob":40000000},"Params":null,"Rewards":null,"ConsumerChains":null,"AssignedKeys":null,"ProviderKeys":null,"ConsumerChainQueueSizes":null,"GlobalSlashQueueSize":null,"RegisteredConsumerRewardDenoms":null,"Proposals":null}}}]
```

Comment:
In here:
```
   {
        "ActionType": "main.relayPacketsAction",
        "Action":
        {
            "ChainA": "provi",
            "ChainB": "democ",
            "Port": "provider",
            "Channel": 0
        },
        "State":
        {
            "democ":
            {
                "ValBalances": null,
                "ValPowers":
                {
                    "alice": 511,
                    "bob": 500,
                    "carol": 500
                },
                "RepresentativePowers": null,
                "Params": null,
                "Rewards": null,
                "ConsumerChains": null,
                "AssignedKeys": null,
                "ProviderKeys": null,
                "ConsumerChainQueueSizes": null,
                "GlobalSlashQueueSize": null,
                "RegisteredConsumerRewardDenoms": null,
                "Proposals": null
            },
            "provi":
            {
                "ValBalances": null,
                "ValPowers":
                {
                    "alice": 511,
                    "bob": 0,
                    "carol": 500
                },
                "RepresentativePowers": null,
                "Params": null,
                "Rewards": null,
                "ConsumerChains": null,
                "AssignedKeys": null,
                "ProviderKeys": null,
                "ConsumerChainQueueSizes": null,
                "GlobalSlashQueueSize": null,
                "RegisteredConsumerRewardDenoms": null,
                "Proposals": null
            }
        }
    },
    {
        "ActionType": "main.relayPacketsAction",
        "Action":
        {
            "ChainA": "provi",
            "ChainB": "democ",
            "Port": "provider",
            "Channel": 0
        },
        "State":
        {
            "democ":
            {
                "ValBalances": null,
                "ValPowers":
                {
                    "alice": 511,
                    "bob": 0,
                    "carol": 500
                },
                "RepresentativePowers": null,
                "Params": null,
                "Rewards": null,
                "ConsumerChains": null,
                "AssignedKeys": null,
                "ProviderKeys": null,
                "ConsumerChainQueueSizes": null,
                "GlobalSlashQueueSize": null,
                "RegisteredConsumerRewardDenoms": null,
                "Proposals": null
            }
        }
    },
```
it seems you `relayPacketsAction` 2 steps in a row and nothing really changes. In the first case the `provi` state is stated as well but otherwise what's the difference?

Reply:
This is just the marshalled democracy steps object.
But to explain what happens, look closely at the democ state - in the first one it has "ValPowers":
                {
                    "alice": 511,
                    "bob": 500,
                    "carol": 500
                },
                
               in the second one 
               
               "ValPowers":
                {
                    "alice": 511,
                    "bob": 0,
                    "carol": 500
                },

Reply:
I missed that. Thanks!